src/v/storage/segment_appender.cc

```diff
@@ -358,6 +358,12 @@ ss::future<> segment_appender::do_next_adaptive_fallocation() {
                    _fallocation_offset,
                    _committed_offset);
                  return _out.allocate(_fallocation_offset, step)
+                   .then([this, step] {
+                       // ss::file::allocate does not adjust logical file size
+                       // hence we need to do that explicitly with an extra
+                       // truncate. This allows for more efficient writes.
```

Comment:
```suggestion
                       // truncate. This allows for more efficient writes.
                       // See https://github.com/redpanda-data/redpanda/pull/18598.
```

Because you put a very good analysis in the cover letter, may make sense to link to the PR?